src/app/components/file-tagging-modal/file-tagging-modal.component.ts

```diff
@@ -0,0 +1,102 @@
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatSnackBar } from '@angular/material';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { CustomErrorStateMatcher } from '../../shared/error-state-matcher';
+import { findTagOptionByCodeset, TagOption } from './tag-option';
+import { defaultSnackbarOptions } from '../../shared/snackbar-options';
+
+@Component({
+  selector: 'file-tagging-modal',
+  templateUrl: './file-tagging-modal.component.html',
+  styleUrls: [
+    './file-tagging-modal.component.scss',
+    '../../../../src/app/shared/modal.component.scss'
+  ],
+})
+export class FileTaggingModal {
+  node: any;
+  isDirectory: boolean;
+  icon: string;
+  name: string;
+  title: string;
+  matcher = new CustomErrorStateMatcher();
+  codeset: number;
+  recursive: boolean = false;
+  tagOptions: TagOption[] = [
+    { title: 'Untagged', codeset: 0, type: 'delete' },
+    { title: 'IBM-1047', codeset: 1047, type: 'text' },
+    { title: 'ISO-8859-1', codeset: 819, type: 'text' },
+    { title: 'Binary', codeset: 65535, type: 'binary' },
+  ];
```

Comment:
You can actually pre-populate tagging info by adding it here, or to the Properties modal

inside `/metadata` call exists `ccsid`
an 819 is like ISO8859-1 
an 1047 is IBM-1047
a -1 is binary
a 0 is untagged
a 1208 is UTF-8

there could be more... those are ones I noticed. 
Also you should be able to tag as UTF-8 no?

Reply:
I added all possible encodings and changes the control to input with autocomplete list.